### PR TITLE
Servicerization of eveything

### PR DESCRIPTION
**Pull request by lexwebb**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,87 +1,38 @@
...
 // Start DB
-const database = new db.DB(config);
-database.connect().then(() => {
+bot.database = new db.DB(bot);
+bot.database.connect().then(() => {
     logger.info('DB Ready');
 }).catch(() => {
     logger.error('Failed to connect to ElasticSearch, terminating process.');
     process.exit(1);
 });
```
Comment (sdf010) [inline app.js]: Can we make starting the services generic?

File: src/http/server.js [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
 const url = require('url');
-const logger = require('winston');
+const ServiceBase = require('../serviceBase');
 
-let listeningPort = 8080;
+exports.HttpServer = class extends ServiceBase {
+    constructor(app) {
+        super(app);
+        this.listeningPort = 8080;
+        this.setPort(this.config.http.port);
```
Comment (sdf010) [inline src/http/server.js]: might be possible to neaten this after merge

File: src/serviceBase.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class ServiceBase {
+    constructor(app) {
+        this.app = app;
+        this.logger = app.logger;
+        this.config = app.config;
+        this.auth = app.auth;
+    }
+}
+
+module.exports = ServiceBase;
```
Comment (sdf010) [inline src/serviceBase.js]: Good work on a Service class

**Review by sdf010 — DISMISSED** [review]
Good work, some changes to make it a bit more neater

File: src/http/server.js [hunk 4]
```diff
@@ -18,8 +18,11 @@ exports.HttpServer = class extends ServiceBase {
             if (urlparameters.pathname === '/easyCalAuth') {
                 const token = urlparameters.query.code;
                 const guild = urlparameters.query.state;
-                bot.logger.info(token);
-                bot.logger.info(guild);
+                bot.database.connect();
```
Comment (sdf010) [inline src/http/server.js]: Do we need to call this again?

File: src/http/server.js [hunk 5]
```diff
@@ -1,32 +1,35 @@
...
-        // TODO: make path read from config
-        if (urlparameters.pathname === '/easyCalAuth') {
-            const token = urlparameters.query.code;
-            const guild = urlparameters.query.state;
-            database.connect();
-            database.putCalendar({
-                id: guild,
-                token,
-            });
+    setPort(port) {
```
Comment (f-doot) [inline src/http/server.js]: Remove this if it's not used

**Review by f-doot — APPROVED** [review]

**Pull request merged**